### PR TITLE
fix(select/telescope): check for nil selection

### PR DESCRIPTION
**Pull request by williamboman**
Hello, thanks for this great plugin! I've been running into some nil access errors when not selecting anything from the telescope menu (for example, [when searching for an entry that doesn't exist so that it clears the results, and then press enter](https://asciinema.org/a/kCM12JgpsuK3ysYAaqNaTvJF2)). Figured that doing a simple check would make sense here

### CODE REVIEW

File: lua/dressing/select/telescope.lua [hunk 1]
```diff
@@ -35,6 +35,10 @@ M.select = function(config, items, opts, on_choice)
       actions.select_default:replace(function()
         local selection = state.get_selected_entry()
         actions._close(prompt_bufnr, false)
+        if not selection then
+          -- User did not select anything.
+          return
```
Comment (stevearc) [inline lua/dressing/select/telescope.lua:40]: To maintain API consistency we'll need to call `on_choice(nil, nil)` before the return

**Review by stevearc — CHANGES_REQUESTED** [review]
Thanks for the PR! One small request and then I'll be happy to merge it

File: lua/dressing/select/telescope.lua (hunk 1 above)
Comment (williamboman) [reply lua/dressing/select/telescope.lua:41]: Done 2f2582e5f716d84d7c9ae77c5439f27e1ac267f7

**Comment by stevearc** [comment]
Awesome, thanks!

**Pull request merged**